internal/agent/device/device.go

```diff
@@ -210,6 +210,12 @@ func (a *Agent) syncDeviceSpec(ctx context.Context) {
 			return
 		}
 
+		// handle policy not ready (download/update)
+		if errors.Is(syncErr, errors.ErrUpdatePolicyNotReady) || errors.Is(syncErr, errors.ErrDownloadPolicyNotReady) {
+			a.handleSyncError(ctx, desired, syncErr)
+			return
```

Comment:
nit: although we are not upgrading yet so to speak the desired state has still been bumped. the expectation is that we rollback before we try again. after rollback the current and desired state will be equal. 

so we still need to rollbackDevice but if we can identify expected cases, perhaps we can handle the log message differently?